hack/.golangci.yaml

```diff
@@ -0,0 +1,12 @@
+---
```

Comment:
maybe we should put a comment at the top pointing to the upstream docs for this config format?

Reply:
this got split out in another PR working on some other class of lints, upon rebase the change in this file should just be adding the new linter 😅